### PR TITLE
Adds support for formatting .sc files using sbt options.

### LINKED ISSUE #1075: REQUEST:  Support for Ammonite scripts
Hi **scalafmt** team :)

I was wondering whether it's possible to format the ammonite repl scripts using `scalafmt`.

I know that there's a slight difference w.r.t entry point of a program in `ammonite` but it would be really nice to have some support for these scripts.

I've a simple script 

```scala

// generate_trimmed_name("PT000033_1.fastq")
def generate_trimmed_name(genome_name:String) : String =
{
  var name_array = genome_name.split("\\.")
  return { name_array(0) + "_trimmed_paired.fastq"}

}


// generate_untrimmed_name("PT000033_1.fastq")
def generate_untrimmed_name(genome_name:String) : String =
{

  var name_array = genome_name.split("\\.")
  return { name_array(0) + "_trimmed_unpaired.fastq"}

}


// velveth_assembly("PT000033", "41")
def velveth_assembly(genome_name:String, k_mer:String) = {

//  velveth PT000033_41 41 -fastq -shortPaired PT000033_1_trimmed_paired.fastq PT000033_2_trimmed_paired.fastq -fastq -short PT000033_1_trimmed_unpaired.fastq PT000033_2_trimmed_unpaired.fastq


  //  var  genome_first = genome_name + "_" + k_mer + ".fastq"

  var  genome_k_mer_name = genome_name + "_" + k_mer
  var  genome_first = genome_name + "_1.fastq"
  var  genome_second = genome_name + "_2.fastq"


  println("velveth " +  genome_k_mer_name +  " " + k_mer + " -fastq -shortPaired " + " " + generate_trimmed_name(genome_first) +  " " + generate_untrimmed_name(genome_first) +  " -fastq -short " + generate_trimmed_name(genome_second) +  " " + generate_untrimmed_name(genome_second))


}




// velvetg_produce_graph("PT000033")
def velvetg_produce_graph(genome_name:String, k_mer:String) = {

//  velvetg PT000033_41 -exp_cov auto -cov_cutoff auto


  var  genome_k_mer_name = genome_name + "_" + k_mer

  println("velvetg " + genome_k_mer_name + " -exp_cov auto -cov_cutoff auto")

}




// assemblathon_stats("PT000033", "41")
def assemblathon_stats(genome_name:String , k_mer:String) = {

  //  assemblathon_stats.pl ./PT000033_41/contigs.fa

  var  genome_k_mer_name = genome_name + "_" + k_mer


  println("assemblathon_stats.pl ./" + genome_k_mer_name + "/contigs.fa")

}

// TODO: 
// abacas.pl -r ../NC000962_3.fasta -q contigs.fa -p promer -b -d -a


/// GENOME ANNOTATION


// prokka_annotation("PT000033", "NC000962_3")
def prokka_annotation(genome_name:String, reference_genome:String) = {

  // cd /home/centos/Module2/PT000033_49

  //  prokka --outdir ./PT000033_prokka --prefix PT000033 contigs.fa_NC000962_3.fasta.fasta

  var genome_prokka_name = genome_name + "_prokka"

  var contigs_reference_genome = "contigs.fa_" + reference_genome + ".fasta.fasta"


  println("prokka --outdir ./" + genome_prokka_name +  " --prefix " + genome_name + " " + contigs_reference_genome)

}


```

This raises the following error with 

```sh

BioDragon/Genesis/ammonite_scripts git/master*
❯ scalafmt
[warn] Error in /Users/eklavya/Projects/ProjectEklavya/Eklavya/Code/BioDragon/Genesis/ammonite_scripts/genesis_module_2.scala: <input>:3: error: expected class or object definition
def generate_trimmed_name(genome_name:String) : String =
^
```

### PR DESCRIPTION
**Pull request by lloydmeta**
Closes #1075 and #574

Took a stab at this.

### CODE REVIEW

File: scalafmt-intellij/src/main/scala/org/scalafmt/intellij/FileDocument.scala [hunk 1]
```diff
@@ -17,13 +17,15 @@ case class FileDocument(project: Option[Project], document: Document) {
...
+    virtualFile.getFileType.getName.compareToIgnoreCase("SC") == 0
   def isScala: Boolean = virtualFile.getFileType.getName == "Scala"
   def canFormat: Boolean = isScala || isSbt
 
   def format(): Unit = if (canFormat) {
     val style = getStyle(project, virtualFile)
     val runner =
-      if (isSbt)
+      if (isSbt || isSc)
         style.runner.copy(dialect = scala.meta.dialects.Sbt0137)
```
Comment (olafurpg) [inline scalafmt-intellij/src/main/scala/org/scalafmt/intellij/FileDocument.scala]: Unrelated to this PR but the line below contains a bug, this line should be
```scala
         style.runner.forSbt
```
to support nice things like trailing commas.
https://github.com/scalameta/scalafmt/blob/542a04500165a8827d2484118526f2c9c3d03835/scalafmt-core/shared/src/main/scala/org/scalafmt/config/ScalafmtRunner.scala#L34

File: scalafmt-intellij/src/main/scala/org/scalafmt/intellij/FileDocument.scala [hunk 2]
```diff
@@ -17,13 +17,15 @@ case class FileDocument(project: Option[Project], document: Document) {
   def path = virtualFile.getPath
   def isSbt: Boolean =
     virtualFile.getFileType.getName.compareToIgnoreCase("SBT") == 0
+  def isSc: Boolean =
+    virtualFile.getFileType.getName.compareToIgnoreCase("SC") == 0
```
Comment (olafurpg) [inline scalafmt-intellij/src/main/scala/org/scalafmt/intellij/FileDocument.scala]: Did you check the file type in IntelliJ for .sc files?

Comment (lloydmeta) [reply scalafmt-intellij/src/main/scala/org/scalafmt/intellij/FileDocument.scala]: Ah good call. I had no idea what I was doing in IntelliJ and just cargo-culted.

Apparently it's a Scala file, so I needed to check for the `.sc` extension.

File: scalafmt-intellij/src/main/scala/org/scalafmt/intellij/FileDocument.scala (hunk 1 above)
Comment (lloydmeta) [reply scalafmt-intellij/src/main/scala/org/scalafmt/intellij/FileDocument.scala]: Done !

**Comment by lloydmeta** [comment]
Tests added in addition to addressing other comments :) Please have another look @olafurpg

File: scalafmt-intellij/src/main/scala/org/scalafmt/intellij/FileDocument.scala [hunk 3]
```diff
@@ -17,14 +17,16 @@ case class FileDocument(project: Option[Project], document: Document) {
...
   def isScala: Boolean = virtualFile.getFileType.getName == "Scala"
   def canFormat: Boolean = isScala || isSbt
 
   def format(): Unit = if (canFormat) {
     val style = getStyle(project, virtualFile)
     val runner =
-      if (isSbt)
-        style.runner.copy(dialect = scala.meta.dialects.Sbt0137)
+      if (isSbt || isSc)
+        style.runner.forSbt
```
Comment (olafurpg) [inline scalafmt-intellij/src/main/scala/org/scalafmt/intellij/FileDocument.scala]: Looking forward to enjoy this fix ^^ been bugged I can't use trailing commas in sbt 1.0 builds

File: scalafmt-tests/src/test/scala/org/scalafmt/cli/CliTest.scala [hunk 4]
```diff
@@ -326,10 +331,15 @@ class CliTest extends AbstractCliTest {
...
       string2dir(
         """|/foo.scala
+           |object    FormatMe
+           |/foo.sc
            |object    FormatMe
            |""".stripMargin
       ),
-      """|/foo.scala
+      """|/foo.sc
+         |object FormatMe
```
Comment (olafurpg) [inline scalafmt-tests/src/test/scala/org/scalafmt/cli/CliTest.scala]: Note that I think package clauses are rejected by the sbt parser, not sure how problematic that is for .sc files.

We'd need to fix the scalameta parser to handle package with sbt dialect.

**Pull request merged**